mlflow/store/dbfs_artifact_repo.py

```diff
@@ -0,0 +1,139 @@
+                                                          'with dbfs:/'
+        self.http_request_kwargs = http_request_kwargs
+
+    def _get_dbfs_path(self, artifact_path):
+        return '/%s/%s' % (strip_prefix(self.artifact_uri, 'dbfs:/'),
+                           strip_prefix(artifact_path, '/'))
+
+    def _get_dbfs_endpoint(self, artifact_path):
+        return "/dbfs%s" % self._get_dbfs_path(artifact_path)
+
+    def log_artifact(self, local_file, artifact_path=None):
+        if artifact_path == '':
+            raise IllegalArtifactPathError('artifact_path cannot be the empty string.')
+        if artifact_path:
+            http_endpoint = self._get_dbfs_endpoint(artifact_path)
+        else:
+            http_endpoint = self._get_dbfs_endpoint(os.path.basename(local_file))
+        with open(local_file, 'r') as f:
+            http_request(endpoint=http_endpoint, method='POST', data=f, **self.http_request_kwargs)
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        if artifact_path:
+            root_http_endpoint = self._get_dbfs_endpoint(artifact_path)
+        else:
+            root_http_endpoint = self._get_dbfs_endpoint(os.path.basename(local_dir))
+        for (dirpath, _, filenames) in os.walk(local_dir):
+            dir_http_endpoint = root_http_endpoint
+            if dirpath != local_dir:
+                rel_path = get_relative_path(local_dir, dirpath)
+                dir_http_endpoint = build_path(root_http_endpoint, rel_path)
+            for name in filenames:
+                endpoint = build_path(dir_http_endpoint, name)
+                with open(build_path(dirpath, name), 'r') as f:
+                    http_request(endpoint=endpoint, method='POST', data=f,
+                                 **self.http_request_kwargs)
+
+    def list_artifacts(self, path=None):
+        if path:
+            dbfs_list_json = {'path': self._get_dbfs_path(path)}
+        else:
+            dbfs_list_json = {'path': self._get_dbfs_path('')}
+        response = _dbfs_list_api(dbfs_list_json, self.http_request_kwargs)
+        json_response = json.loads(response.text)
+        # /api/2.0/dbfs/list will not have the 'files' key in the response for empty directories.
+        infos = []
+        artifact_prefix = strip_prefix(self.artifact_uri, 'dbfs:')
+        if json_response.get('error_code', None) == RESOURCE_DOES_NOT_EXIST:
+            return []
+        dbfs_files = json_response.get('files', [])
+        for dbfs_file in dbfs_files:
+            is_dir = dbfs_file['is_dir']
+            artifact_size = None if is_dir else dbfs_file['file_size']
+            stripped_path = strip_prefix(dbfs_file['path'], artifact_prefix)
+            infos.append(FileInfo(stripped_path, is_dir, artifact_size))
+        return sorted(infos, key=lambda f: f.path)
+
+    def download_artifacts(self, artifact_path):
+        with TempDir(remove_on_exit=False) as tmp:
```

Comment:
Here we could use `tempfile.mkdtemp()` but I think this is fine for consistency (we should refactor some of the common logic out of the DBFS/Azure/S3/GCS artifact stores at some point & having common code now seems like it'll make the refactor easier later on)